### PR TITLE
chore: remove old io

### PR DESCRIPTION
**Pull request by dranikpg**
Remove old IO
* Remove SinkReplyBuilder, RedisReplyBuilder, MCReplyBuilder, replace them with new __2 version
* Remove MGetResponse and SendScoredArray 
* Simplify to use new small interface
    * CaptruingReplyBuilder
    * HttpReplyBuilder
    * InterpreterReplier

### CODE REVIEW

**Comment by dranikpg** [comment]
I lost your changes from #3946 while deleting old code, need to add them into the new

File: src/server/string_family.cc
Comment (romange) [inline src/server/string_family.cc:561]: what's 27? :)

**Comment by romange** [comment]
Please also run the regtests on it. Please do not merge it after the approval, I would like to cut 1.25 early next week without this PR.

File: src/server/string_family.cc
Comment (dranikpg) [reply src/server/string_family.cc:561]: My birthday date 🙂 Kidding, good that you found it, forgot to add a todo. To prevent sso, because we have a self referential struct, changed it to a `unique_ptr<char[]>`

**Review by romange — APPROVED** [review]

**Pull request merged**
